.github/workflows/d.yml

```
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: D
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!dependabot/**'
+    tags:
+      - '**'
+    paths:
+      - '.dockerignore'
+      - '.github/workflows/d.yml'
+      - 'ci/docker/*d*'
```

All `.dockerfile`s are matched...

---

@kou,
I have not listed the `d/` files in `.dockerignore`.
One question: is docker_ruby already included in c_glib?

Previously the CI/CD section, I just attempted adapting swift-lang recipe.

---

```suggestion
      - 'ci/docker/*-d.dockerfile'
```

>  One question: is docker_ruby already included in c_glib?

What is referred by the `docker_ruby`? https://github.com/apache/arrow/blob/main/ci/docker/linux-apt-ruby.dockerfile ?